lib/ClangImporter/SwiftLookupTable.cpp

```diff
@@ -2029,6 +2029,20 @@ void importer::addMacrosToLookupTable(SwiftLookupTable &table,
   }
 }
 
+static bool isForwardDecl(clang::Decl *decl) {
+  if (auto classDecl = dyn_cast<clang::ObjCInterfaceDecl>(decl))
+    return !classDecl->isThisDeclarationADefinition();
+  if (auto protoDecl = dyn_cast<clang::ObjCProtocolDecl>(decl))
+    return !protoDecl->isThisDeclarationADefinition();
+  if (auto tagDecl = dyn_cast<clang::TagDecl>(decl))
+    return !tagDecl->isThisDeclarationADefinition();
+
+  // FIXME: There are other AST nodes that can be forward decls, but our current
+  // use site doesn't care about those.
+
+  return false;
+}
```

Comment:
This is `ClangImporter::Implementation::shouldIgnoreBridgeHeaderTopLevelDecl`. We could hand the table finalizer a const ref to the callers.

Reply:
Good catch. That method actually is just trying to see if the declaration is a forward decl, so I think I'll just refactor it into this:
```cpp
bool importer::isForwardDeclOfType(clang::Decl *D) {
  if (auto *ID = dyn_cast<clang::ObjCInterfaceDecl>(D)) {
    // ...existing implementation of `shouldIgnoreBridgeHeaderTopLevelDecl()` omitted...
  return false;
}

bool ClangImporter::Implementation::shouldIgnoreBridgeHeaderTopLevelDecl(
    clang::Decl *D) {
  return importer::isForwardDeclOfType(D);
}
```

And then call the free function from SwiftLookupTable.cpp, rather than trying to get a ClangImporter::Implementation into SwiftLookupTable